op-supervisor/supervisor/backend/backend.go

```diff
@@ -134,21 +174,23 @@ func (su *SupervisorBackend) initResources(ctx context.Context, cfg *config.Conf
 		}
 	}
 
+	eventOpts := event.DefaultRegisterOpts()
 	// initialize all cross-unsafe processors
 	for _, chainID := range chains {
 		worker := cross.NewCrossUnsafeWorker(su.logger, chainID, su.chainDBs)
-		su.crossUnsafeProcessors.Set(chainID, worker)
+		su.eventSys.Register(fmt.Sprintf("cross-unsafe-%s", chainID), worker, eventOpts)
```

Comment:
Did this previously disallow replacing the working (ie registering the same chain ID twice)? And if so does the evenSys preserve that behaviour?

Reply:
The chains returned by the dependency set util are always unique. And this is in the initialization path. So there shouldn't be existing cross-unsafe-workers. There's a `panic` in the `Register` func if the name already exists, so if there's some programming error there we'd catch it before the supervisor fully comes online.